RecoEgamma/EgammaTools/interface/EnergyScaleCorrection.h

```diff
@@ -193,8 +194,8 @@ class EnergyScaleCorrection {
 
   //data members
   FileFormat smearingType_;
-  std::vector<std::pair<CorrectionCategory, ScaleCorrection> > scales_;
-  std::vector<std::pair<CorrectionCategory, SmearCorrection> > smearings_;
+  std::map<CorrectionCategory, ScaleCorrection> scales_;
+  std::map<CorrectionCategory, SmearCorrection> smearings_;
```

Comment:
why not unordered_map?

Reply:
This idea crossed my mind.
However, it looked like some of the logic implied that the key ordering is important.
I didn't make enough effort to prove this wrong to myself.

Reply:
Oh, also, I'm was not sure how to write a hash that would return the right value starting from `{2 ints, 3 doubles}`, which would correspond to a range of values mapping on the same key.